app/data_client.go

```diff
@@ -471,6 +483,16 @@ func convertBsonToNative(data interface{}) interface{} {
 	}
 }
 
+func datasetFromProto(dataset *setPb.Dataset) *Dataset {
+	timeCreated := dataset.TimeCreated.AsTime()
```

Comment:
`dataset.TimeCreated` is a pointer, right? We should probably check for nil values here.